### PR TITLE
[Merged by Bors] - feat (Algebra/Order/AddTorsor) : OrderedVAdd, OrderedCancelVAdd

### PR DESCRIPTION
**Pull request by ScottCarnahan**
This PR introduces prop-valued mixins for vector addition operations that preserve or reflect order, and provides instances relating it to ordered commutative monoids and to covariant/contravariant classes.  In a future PR, we expect to use this to generalize Hahn module structures, for example giving an `R((z))`-module structure to `z^s V((z))` for `V` an `R`-module.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/AddTorsor.lean [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    exact (OrderedCancelVAdd.le_of_vadd_le_vadd_right b a c h.ge)
+
+instance OrderedCancelAddCommMonoid.toOrderedCancelVAdd [OrderedCancelAddCommMonoid G] :
+    OrderedCancelVAdd G G where
+  le_of_vadd_le_vadd_left _ _ _ := le_of_add_le_add_left
+  le_of_vadd_le_vadd_right _ _ _ := le_of_add_le_add_right
+
+namespace VAdd
+
+theorem vAdd_lt_vAdd_of_le_of_lt [LE G] [Preorder P] [OrderedCancelVAdd G P]
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/AddTorsor.lean:81]: The first group of caps gets lowercased
```suggestion
theorem vadd_lt_vadd_of_le_of_lt [LE G] [Preorder P] [OrderedCancelVAdd G P]
```
and elsewhere

**Review by YaelDillies — COMMENTED** [review]
Sorry I am going to tank your dreams but please do not follow one-type typeclasses (eg `OrderedAddMonoid`) when writing two-type typeclasses (eg `OrderedVAdd`). The reason is that the number of algebraic order interactions is already pretty high on a single type `α`, so it becomes unbearable when you add a second type `β` (and you can guess since you didn't define `OrderedAddTorsor`, which is the object you're supposedly interested in).

Instead, we handle that using prop-valued mixins. Here, it would be akin to `PosSMulMono`.

Final note: People do care about multiplicative torsors sometimes, so it would be nice if you wrote all the theory multiplicatively and used `to_additive` to get back what you wanted in the first place.

**Comment by ScottCarnahan** [comment]
Thank you for letting me know.  I will rewrite the classes as prop-valued mixins.

**Review by jcommelin — COMMENTED** [review]
It would be nice to have this multiplicativized at some point. But let's move forward with this for now.

bors merge

**Pull request closed without merging**